clang/include/clang/Sema/Sema.h

```diff
@@ -1759,6 +1759,34 @@ class Sema final {
     }
   };
 
+  /// Bitmask to contain the list of reasons a single diagnostic should be
+  /// emitted, based on its language.  This permits multiple offload systems
+  /// to coexist in the same translation unit.
+  enum class DeviceDiagnosticReason : unsigned char {
```

Comment:
```suggestion
  enum class DeviceDiagnosticReason {
```
Size doesn't matter in this case, may as well go with the default `int`.

Reply:
Done